addons/spreadsheet/static/src/chart/plugins/odoo_chart_ui_plugin.js

```diff
@@ -133,6 +134,24 @@ export class OdooChartUIPlugin extends OdooUIPlugin {
     // Private
     // -------------------------------------------------------------------------
 
+    /**
+     * Determine grouping based on active date filters
+     * @private
+     */
+    _getDynamicGroupBy() {
+        const dateFilter = this.getters.getGlobalFilters().filter(f => f.type === "date")[0];
+        const value = dateFilter ? this.getters.getGlobalFilterValue(dateFilter.id) : "";
+        const months = monthsOptions.map(m => m.id);
+        if (dateFilter?.rangeType === "fixedPeriod") {
+            if(months.includes(value?.period)) {
+                return ["date:day"];
+            }
+            else {
+                return ["date:month"];
+            }
+        }
```

Comment:
i'm absolutely sure that it's not a "global" solution.... it only works in your case and will not work for anything else. please don't change something global for a specific use case without having a good **good** talk with us and our PO (@Francois-frgi)